src/functions/Coverage.ps1

```diff
@@ -1122,9 +1122,18 @@ function Start-TraceScript ($Breakpoints) {
     }
 
     if (-not $registered) {
-        $patched = $true
-        [Pester.Tracing.Tracer]::Patch($PSVersionTable.PSVersion.Major, $ExecutionContext, $host.UI, $tracer)
-        Set-PSDebug -Trace 1
+
+        # detect if code coverage is enabled throuh Pester tracer, and in that case just add us as a second tracer
+        if ([Pester.Tracing.Tracer]::ShouldRegisterTracer($tracer, <# overwrite: #> $false)) {
+            $patched = $false
+            $registered = $true
+            [Pester.Tracing.Tracer]::Register($tracer)
+        }
+        else {
+            $patched = $true
```

Comment:
> so I just needed to check in patch and unpatch not just for profiler tracer, but also for pester tracer.

@fflaten  this is what I meant by that. I should hide this under a Env variable probably, so we don't get into broken state after cancellation, as you pointed out.

Reply:
See few lines of the unchanged code above, it then makes more sense imho.